### PR TITLE
add context tokens to api response

### PR DESCRIPTION
**Pull request by CornHusker89**

### CODE REVIEW

File: galaxygpt-api/Program.cs [hunk 1]
```diff
@@ -127,8 +127,10 @@ public static void Main(string[] args)
                     Context = context.Item1,
                     Duration = requestStart.ElapsedMilliseconds.ToString(),
                     Version = version,
-                    QuestionTokens = context.Item2.ToString(),
-                    ResponseTokens = answer.Item2.ToString()
+                    PromptTokens = answer.Item2.ToString(),
+                    ContextTokens = context.Item2.ToString(),
+                    QuestionTokens = context.Item3.ToString(),
+                    ResponseTokens = answer.Item3.ToString()
```
Comment (smallketchup82) [inline galaxygpt-api/Program.cs:133]: yeah no this shit is completely unreadable. switch to friendly names

**Comment by smallketchup82** [comment]
as an added fyi, we can probably skip tokenizing stuff ourselves and can probably use openai-dotnet

https://github.com/openai/openai-dotnet/blob/main/src/Custom/Chat/ChatTokenUsage.cs

should be available in the return of `CompleteChatAsync` i think

File: galaxygpt/AiClient.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ public partial class AiClient(
 
         string finalMessage = messages[^1].Content[0].Text;
         await ModerateText(finalMessage, moderationClient);
-        return (finalMessage, gptTokenizer.CountTokens(finalMessage));
+        return (finalMessage, gptTokenizer.CountTokens(messages.First().Content[0].Text), clientResult.Value.Usage.OutputTokenCount);
```
Comment (smallketchup82) [inline galaxygpt/AiClient.cs:76]: this breaks tests (mocked clientResult doesnt have a value for OutputTokenCount most likely).

**Review by smallketchup82 — CHANGES_REQUESTED** [review]

**Comment by smallketchup82** [comment]
looks good to me otherwise

File: galaxygpt/AiClient.cs (hunk 2 above)
Comment (CornHusker89) [reply galaxygpt/AiClient.cs:76]: its actually caused either by `gptTokenizer.CountTokens(messages.First().Content[0].Text)` or `clientResult.Value.Usage.OutputTokenCount`, being null, i dont know which one and i wont really be able to test this today

Comment (smallketchup82) [reply galaxygpt/AiClient.cs:76]: its the latter. you dont have to change anything here, you'll need to change tests to provide a value for OutputTokenCount

**Review by smallketchup82 — APPROVED** [review]

**Pull request merged**